nixos/modules/hardware/video/intel.nix

```diff
@@ -0,0 +1,13 @@
+{ config, lib, pkgs, ... }:
+{
+  options.hardware.intel.opencl.enable = lib.mkEnableOption (lib.mdDoc
+    "intel opencl runtime (Install icd for Intel GPUs)"
+  );
+
+  config = lib.mkIf config.hardware.intel.opencl.enable {
+    hardware.opengl.extraPackages = with pkgs; [
+      intel-compute-runtime
+      intel-ocl
```

Comment:
Having that under video is not fully correct, as it also has parts that only use the CPU, it is more general intel hardware.